packages/utilities/dom/src/query.ts

```diff
@@ -1,3 +1,9 @@
+type Ctx<T extends string> = Partial<Record<T, () => Document | ShadowRoot | Node>>
```

Comment:
Let's move this closer to the withRootHelper function, and simplify.

```suggestion
type Ctx = { getRootNode?: () => Document | ShadowRoot | Node }
```